File: web/src/types/opportunity.ts
export type OpportunityType = 'Deal' | 'Cashback' | 'Bonus' | 'Rebate' | 'PriceAlert'
export type OpportunityStatus = 'New' | 'Seen' | 'Saved' | 'ActedOn' | 'Dismissed' | 'Expired'
export type ConfidenceLevel = 'High' | 'Medium' | 'Low'

export interface Opportunity {
  id: string
  source: string
  sourceId: string
  type: OpportunityType
  status: OpportunityStatus
  title: string
  description: string | null
  value: number | null
  originalValue: number | null
  discountPercent: number | null
  url: string
  imageUrl: string | null
  expiresAt: string | null
  confidence: ConfidenceLevel
  tags: string[]
  metadata: Record<string, unknown>
  createdAt: string
  updatedAt: string
}

File: src/DealMe.Core/Domain/Entities/Opportunity.cs
namespace DealMe.Core.Domain.Entities;

using DealMe.Core.Domain.Enums;

public class Opportunity
{
    public Guid Id { get; set; }
    public string Source { get; set; } = string.Empty;
    public string SourceId { get; set; } = string.Empty;
    public OpportunityType Type { get; set; }
    public OpportunityStatus Status { get; set; } = OpportunityStatus.New;
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal? Value { get; set; }
    public decimal? OriginalValue { get; set; }
    public int? DiscountPercent { get; set; }
    public string Url { get; set; } = string.Empty;
    public string? ImageUrl { get; set; }
    public DateTimeOffset? ExpiresAt { get; set; }
    public ConfidenceLevel Confidence { get; set; }
    public string Tags { get; set; } = "[]";
    public string Metadata { get; set; } = "{}";
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }
}
